base/compiler/stmtinfo.jl

```diff
@@ -7,8 +7,7 @@ to re-consult the method table. This info is illegal on any statement that is
 not a call to a generic function.
 """
 struct MethodMatchInfo
```

Comment:
Maybe this type should be combined with `MethodLookupResult`?

Reply:
I'm not quite sure yet. There's some additional information about method matches I may want to keep in the near future, that I think could go here nicely. My preference would be to keep this as is for now and if it turns out that information would be better somewhere else, we can always refactor this, since it's all internal.